src/backend/BSSched/BuildJob.pm

```diff
@@ -546,6 +546,11 @@ sub jobfinished {
     BSUtil::appendstr("$dst/logfile", "\nRetried build at ".localtime(time())." returned same result, skipped\n");
     my $jobhist = makejobhist($info, $status, $js, 'unchanged');
     addbuildstats($jobdatadir, $dst, $jobhist) if $all{'_statistics'};
+
+    my $ccachetar = "$jobdatadir/_ccache.tar";
+    my $occachetar = "$dst/_ccache.tar";
+    rename($ccachetar, "$dst/_ccache.tar") if -e $ccachetar && !-e $occachetar;
+
```

Comment:
You should also use `$occachetar` in the rename() call. And maybe use $all{'_ccache.tar'} instead of the -e?